src/app/my-account/my-account.component.ts

```diff
@@ -17,19 +17,22 @@ import {HttpErrorResponse} from "@angular/common/http";
   styleUrls: ['./my-account.component.scss'],
   providers: [DatePipe]
 })
-export class MyAccountComponent implements OnInit {
+export class MyAccountComponent implements OnDestroy, OnInit {
   public person: Person;
 
   public paymentMethods: PaymentMethod[]|undefined = undefined;
   registerErrorDescription: string | undefined;
   registerSucessMessage: string | undefined;
   protected readonly faExclamationTriangle = faExclamationTriangle;
 
+  private savedCardsTimer: ReturnType<typeof setTimeout>; // https://stackoverflow.com/a/56239226
```

Comment:
```suggestion
  private savedCardsTimer: undefined | ReturnType<typeof setTimeout>; // https://stackoverflow.com/a/56239226
```

Since we only conditionally set this in OnInit maybe worth making undefined part of the type, or initialising to null here.